iceoryx_posh/include/iceoryx_posh/internal/popo/sample.inl

```diff
@@ -24,116 +24,125 @@ namespace popo
-template <typename S, typename>
-inline T* Sample<T>::operator->() noexcept
+template <typename T, typename H>
+inline T* Sample<T, H>::operator->() noexcept
 {
     return get();
 }
 
-template <typename T>
-inline const T* Sample<T>::operator->() const noexcept
+template <typename T, typename H>
+inline const T* Sample<T, H>::operator->() const noexcept
 {
     return get();
 }
 
-template <typename T>
-template <typename S, typename>
-inline S& Sample<T>::operator*() noexcept
+template <typename T, typename H>
+inline T& Sample<T, H>::operator*() noexcept
 {
     return *get();
 }
 
-template <typename T>
-inline const T& Sample<T>::operator*() const noexcept
+template <typename T, typename H>
+inline const T& Sample<T, H>::operator*() const noexcept
 {
     return *get();
 }
 
-template <typename T>
-inline Sample<T>::operator bool() const noexcept
+template <typename T, typename H>
+inline Sample<T, H>::operator bool() const noexcept
 {
     return get() != nullptr;
 }
 
-template <typename T>
-template <typename S, typename>
-inline T* Sample<T>::get() noexcept
+template <typename T, typename H>
+inline T* Sample<T, H>::get() noexcept
 {
     return m_members.sampleUniquePtr.get();
 }
 
-template <typename T>
-inline const T* Sample<T>::get() const noexcept
+template <typename T, typename H>
+inline const T* Sample<T, H>::get() const noexcept
 {
     return m_members.sampleUniquePtr.get();
 }
 
-template <typename T>
-template <typename S, typename>
-inline mepoo::ChunkHeader* Sample<T>::getHeader() noexcept
+template <typename T, typename H>
+inline typename Sample<T, H>::ConditionalConstChunkHeader_t* Sample<T, H>::getHeader() noexcept
 {
     return mepoo::ChunkHeader::fromPayload(m_members.sampleUniquePtr.get());
 }
 
-template <typename T>
-inline const mepoo::ChunkHeader* Sample<T>::getHeader() const noexcept
+template <typename T, typename H>
+inline const mepoo::ChunkHeader* Sample<T, H>::getHeader() const noexcept
 {
     return mepoo::ChunkHeader::fromPayload(m_members.sampleUniquePtr.get());
 }
 
-template <typename T>
+template <typename T, typename H>
+template <typename R, typename>
+inline R& Sample<T, H>::getCustomHeader() noexcept
+{
+    return *mepoo::ChunkHeader::fromPayload(m_members.sampleUniquePtr.get())->template customHeader<R>();
+}
+
+template <typename T, typename H>
+template <typename R, typename>
+inline const R& Sample<T, H>::getCustomHeader() const noexcept
+{
+    return *mepoo::ChunkHeader::fromPayload(m_members.sampleUniquePtr.get())->template customHeader<R>();
+}
+
+template <typename T, typename H>
 template <typename S, typename>
-inline void Sample<T>::publish() noexcept
+inline void Sample<T, H>::publish() noexcept
 {
     if (m_members.sampleUniquePtr)
     {
         m_members.publisherRef.get().publish(std::move(*this));
     }
-
     else
     {
-        /// @todo Notify caller of attempt to publish invalid chunk. Or something ?
+        LogWarn() << "Tried to publish empty sample!";
```

Comment:
I was not quite sure if logging is enough or if it maybe should be an `LogError`. What do you think?

Reply:
First I thought this can only happen when the developer tries to publish an already moved or published sample. In this case we could write "Trying to publish already published or moved sample!". But we could also construct an empty sample :( ... and this could also occur.

I would you should print a `LogError` message here and call additionally the errorHandler with a moderate warning since nothing gets corrupted here but it could indicate a kaputt algorithm.